system/services/formbuilder/FormBuilderFilterService.cfc

```diff
@@ -0,0 +1,1095 @@
+/**
+ * Provides logic around filters in form builder
+ *
+ * @autodoc
+ * @singleton
+ * @presideservice
+ */
+component {
+
+// CONSTRUCTOR
+	/**
+	* @formBuilderService.inject                 formBuilderService
+	* @rulesEngineConditionService.inject        rulesEngineConditionService
+	 */
+
+	 public any function init(
+	 	  required any formBuilderService
+	 	, required any rulesEngineConditionService
+	 ) {
+		_setFormBuilderService( arguments.formBuilderService );
+		_setRulesEngineConditionService( arguments.rulesEngineConditionService );
+
+		return this;
+	}
+
+	public boolean function userQuestionResponseMatchesCondition(
+		  string conditionId
+		, string formId
+		, string submissionId
+	) {
+
+		if ( !$isWebsiteUserLoggedIn() ) {
+			return false;
+		}
+
+		if ( !len( trim( arguments.conditionId ) ) ) {
+			return false;
+		}
+
+
+		var match = _getRulesEngineConditionService().evaluateCondition(
```

Comment:
Stylisticly here, I think we can just return the result of the function call, rather than storing in a variable. i.e.

```
// this
return someFunctionCall( ... );

// not this
var result = someFunctionCall( ... );
return result;
```

Unless we do something with result before returning it.

Reply:
I'd removed a writeDump(match) and should have changed it back to returning the result immediately